python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,346 @@
+    data in multiple bands that are stored as separate objects.
+    Analyzing the data can be easier using a Multiband object that
+    wraps the underlying data as a single data cube that can be sliced and
+    updated as a single object.
+
+    `MultibandBase` is designed to contain the most important universal
+    methods for initializing, slicing, and extracting common parameters
+    (such as the bounding box or XY0 position) to all of the single band classes.
+    """
+    def __init__(self, singles, slices=None, **kwargs):
+        """Initialize a `MultibandBase` object
+
+        Must be overloaded in derived classes.
+        At a minimum, the derived class must be initialized with
+        a `_singles` attribute that is a list of single band objects,
+        a `_filters` attribute that is a list of filter
+        names, a `_bbox` attribute is a `Box2I`,
+        and a `_xy0` attribute that is a `Point2I`.
+
+        Below are the only required `__init__` parameters for all
+        dervied classes.
+
+        Parameters
+        ----------
+        singles: list, `OrderedDict`, or `MultibandBase`
+            Either a list of single band objects or an `OrderedDict` with
+            filter names as keys and single band objects as values, or
+            an instance of an object that inherits from `MultibandBase`.
+        slices: list
+            Slices used to extract a subset of an image
+            (used when `singles` inherits from `MultibandBase` to
+            create a new object with a slice of the original).
+        """
+        err = "Inherited classes from `MultibandBase` require an `__init__` method"
+        raise NotImplementedError(err)
+
+    @property
+    def filters(self):
+        """List of filter names for the single band objects
+        """
+        return self._filters
+
+    @property
+    def singles(self):
+        """List of afw single band objects
+        """
+        return self._singles
+
+    @property
+    def bbox(self):
+        """Bounding box
+        """
+        return self._bbox
+
+    def getBBox(self):
+        """Get the bounding box
+        """
+        return self._bbox
+
+    @property
+    def XY0(self):
+        """Minimum coordinate in the bounding box
+        """
+        return self._xy0
+
+    def setXY0(self, xy0):
+        """Update the XY0 position for each single band object
+        """
+        assert isinstance(xy0, Point2I)
+        self._xy0 = xy0
+        for n in range(len(self.singles)):
+            self.singles[n].setXY0(xy0)
+
+    @property
+    def x0(self):
+        """X0
+
+        X component of XY0 `Point2I.getX()`
+        """
+        return self.XY0.getX()
+
+    @property
+    def y0(self):
+        """Y0
+
+        Y component of XY0 `Point2I.getY()`
+        """
+        return self.XY0.getY()
+
+    @property
+    def yx0(self):
+        """Minimum (y,x) position
+        """
+        return (self.y0, self.x0)
```

Comment:
I'm not a huge fan of having a reversed version of xy0 in this form:
 - the symmetry with `xy0` in terms of naming would imply that it returns some kind of point object
 - anything defined here should also be available on single-band Image/MaskedImage/Exposure objects.

I'm guessing you want it to offset indices into numpy array views?  If so, I'm not opposed to having something that does this, but I'd like a name that suggests its purpose ("arrayOrigin"?) rather than symmetry with `xy0`, and whatever you go with should be added to the single-band classes.